File: app/src/main/java/com/dss886/dotaautochess/utils/UIUtils.java
package com.dss886.dotaautochess.utils;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by dss886 on 2019/1/26.
 */
public class UIUtils {

    public static final int KEEP = -3;

    public static int getScreenWidth(Context context) {
        if (context == null) {
            return 0;
        } else {
            DisplayMetrics dm = context.getResources().getDisplayMetrics();
            return dm == null ? 0 : dm.widthPixels;
        }
    }

    public static int dp2px(Context context, int dp) {
        float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dp * scale + 0.5F);
    }

    public static void updateLayout(View view, int w, int h) {
        if (view != null) {
            ViewGroup.LayoutParams params = view.getLayoutParams();
            if (params != null && (params.width != w || params.height != h)) {
                if (w != KEEP) {
                    params.width = w;
                }
                if (h != KEEP) {
                    params.height = h;
                }
                view.setLayoutParams(params);
            }
        }
    }

    public static void updateLayoutMargin(View view, int l, int t, int r, int b) {
        if (view != null) {
            ViewGroup.LayoutParams params = view.getLayoutParams();
            if (params != null) {
                if (params instanceof ViewGroup.MarginLayoutParams) {
                    ViewGroup.MarginLayoutParams marginParams = ((ViewGroup.MarginLayoutParams) params);
                    if (l != KEEP) {
                        marginParams.leftMargin = l;
                    }
                    if (t != KEEP) {
                        marginParams.topMargin = t;
                    }
                    if (r != KEEP) {
                        marginParams.rightMargin = r;
                    }
                    if (b != KEEP) {
                        marginParams.bottomMargin = b;
                    }
                    view.setLayoutParams(params);
                }
            }
        }
    }

    public static int getColorBrightness(int color) {
        return ((color & 0xff) * 19595
                + ((color >> 8) & 0xff) * 38469
                + ((color >> 16) & 0xff) * 7472) >> 16;
    }

    public static void setVisibility(View view, int visibility) {
        if (view != null && view.getVisibility() != visibility) {
            view.setVisibility(visibility);
        }
    }

}

File: app/src/main/java/com/dss886/dotaautochess/utils/UIUtils.kt
package com.dss886.dotaautochess.utils

import android.content.Context
import android.view.View
import android.view.ViewGroup

/**
 * Created by dss886 on 2019/1/26.
 */
object UIUtils {

    const val KEEP = -3

    fun getScreenWidth(context: Context?): Int {
        return if (context == null) 0 else {
            val dm = context.resources.displayMetrics
            dm?.widthPixels ?: 0
        }
    }

    fun updateLayout(view: View?, w: Int, h: Int) {
        if (view != null) {
            val params = view.layoutParams
            if (params != null && (params.width != w || params.height != h)) {
                if (w != KEEP) {
                    params.width = w
                }
                if (h != KEEP) {
                    params.height = h
                }
                view.layoutParams = params
            }
        }
    }

    fun updateLayoutMargin(view: View?, l: Int, t: Int, r: Int, b: Int) {
        if (view != null) {
            val params = view.layoutParams
            if (params != null) {
                if (params is ViewGroup.MarginLayoutParams) {
                    if (l != KEEP) {
                        params.leftMargin = l
                    }
                    if (t != KEEP) {
                        params.topMargin = t
                    }
                    if (r != KEEP) {
                        params.rightMargin = r
                    }
                    if (b != KEEP) {
                        params.bottomMargin = b
                    }
                    view.layoutParams = params
                }
            }
        }
    }

    fun getColorBrightness(color: Int): Int {
        return ((color and 0xff) * 19595
                + (color shr 8 and 0xff) * 38469
                + (color shr 16 and 0xff) * 7472) shr 16
    }

    fun setVisibility(view: View?, visibility: Int) {
        if (view != null && view.visibility != visibility) {
            view.visibility = visibility
        }
    }

}
